src/snapshot.rs

```diff
@@ -235,12 +288,35 @@ impl<'a, 'b> Snapshot<'a, 'b> {
         result
     }
 
+    /// Check disk usage of the destination
+    ///
+    /// NOTE: This requires `Image` because we want to ensure this is called
+    /// after the file is created.
+    #[cfg(target_family = "unix")]
+    fn check_disk_usage(&self, _: &Image) -> Result<()> {
+        disk_usage::check(
+            self.destination,
+            &self.memory_ranges,
+            self.max_disk_usage,
+            self.max_disk_usage_percentage,
+        )
+    }
+
+    /// Check disk usage of the destination
+    ///
+    /// On non-Unix platforms, this operation is a no-op.
+    #[cfg(not(target_family = "unix"))]
+    fn check_disk_usage(&self, _: &Image) -> Result<()> {
+        Ok(())
```

Comment:
Are we doing anything to mitigate Windows users thinking the new CLI options have an effect?

A few ideas:

1. use conditional compilation via `#[cfg(...)]` to exclude the CLI options on Windows
2. ensure the `--help` output indicates they have no effect
3. log a warning in this function indicating that some specified CLI options are being ignored
4. implement an actual check for Windows too

Reply:
This code isn't used on non-unix platforms as is, but I understand the potential for concern.  Eventually, I'd like to refactor this part out such this code isn't linked at all said platforms.  For now, I've updated the code to error if the user tries to check disk usage on non-unix platforms.

Does the update address your concern?

Reply:
Yes, thank you!